Compiler Warning: "using integer absolute value function 'abs' when argument is of floating point type"

# The Warning

The following warning occurred while building Drake using OSX 10.11.6, clang Apple LLVM version 7.3.0, and `cmake` / `ninja`: 

```
[101/253] Building CXX object multibody/benchmarks/cylinder_torque_free_analytical_solution/te...les/cylinder_torque_free_analytical_solution.dir/cylinder_torque_free_analytical_solution.cc.o
/Users/liang/dev/drake-distro-4/drake/multibody/benchmarks/cylinder_torque_free_analytical_solution/test/cylinder_torque_free_analytical_solution.cc:119:31: warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]
  const double quat_epsilon = abs(1.0 - quat.norm());
                              ^
/Users/liang/dev/drake-distro-4/drake/multibody/benchmarks/cylinder_torque_free_analytical_solution/test/cylinder_torque_free_analytical_solution.cc:158:7: note: in instantiation of function template specialization 'drake::benchmarks::cylinder_torque_free_analytical_solution::TestQuaternionDtConstraintFromQuaternionDt<double>' requested here
      TestQuaternionDtConstraintFromQuaternionDt(quat_AB, quatDt);
      ^
/Users/liang/dev/drake-distro-4/drake/multibody/benchmarks/cylinder_torque_free_analytical_solution/test/cylinder_torque_free_analytical_solution.cc:613:7: note: in instantiation of function template specialization 'drake::benchmarks::cylinder_torque_free_analytical_solution::CalculateAngularVelocityExpressedInBFromQuaternionDt<double>' requested here
      CalculateAngularVelocityExpressedInBFromQuaternionDt(
      ^
```

--- a/drake/multibody/benchmarks/cylinder_torque_free_analytical_solution/test/cylinder_torque_free_analytical_solution.cc
+++ b/drake/multibody/benchmarks/cylinder_torque_free_analytical_solution/test/cylinder_torque_free_analytical_solution.cc
@@ -112,7 +112,8 @@ T CalculateQuaternionDtConstraintFromQuaternionDt(
 template <typename T>
 bool TestQuaternionDtConstraintFromQuaternionDt(
     const Eigen::Quaternion<T>& quat, const Vector4<T>& quatDt) {
-
+  using std::abs;
+  
   // For an accurate test, the quaternion should be reasonably accurate.
   const double double_epsilon = std::numeric_limits<double>::epsilon();
   const double tolerance = 800 * double_epsilon;

--- a/drake/multibody/benchmarks/cylinder_torque_free_analytical_solution/test/cylinder_torque_free_analytical_solution.cc
+++ b/drake/multibody/benchmarks/cylinder_torque_free_analytical_solution/test/cylinder_torque_free_analytical_solution.cc
@@ -113,7 +113,7 @@ template <typename T>
 bool TestQuaternionDtConstraintFromQuaternionDt(
     const Eigen::Quaternion<T>& quat, const Vector4<T>& quatDt) {
   using std::abs;
-  
+
   // For an accurate test, the quaternion should be reasonably accurate.
   const double double_epsilon = std::numeric_limits<double>::epsilon();
   const double tolerance = 800 * double_epsilon;

